test/e2e/deploy/volumeleaks.go

```diff
@@ -44,15 +46,28 @@ func GetHostVolumes(d *Deployment) map[string][]string {
 	// script appears to be "no such file".
 	for worker := 1; ; worker++ {
 		sshcmd := fmt.Sprintf("%s/_work/%s/ssh.%d", os.Getenv("REPO_ROOT"), os.Getenv("CLUSTER"), worker)
-		ssh := exec.Command(sshcmd, cmd)
-		// Intentional Output instead of CombinedOutput to dismiss warnings from stderr.
-		// lvs may emit lvmetad-related WARNING msg which can't be silenced using -q option.
-		out, err := ssh.Output()
-		if err != nil && os.IsNotExist(err) {
+		if _, err := os.Stat(sshcmd); err == nil {
+			for retry = 1; retry < 4; retry++ {
+				ssh := exec.Command(sshcmd, cmd)
+				// Intentional Output instead of CombinedOutput to dismiss warnings from stderr.
+				// lvs may emit lvmetad-related WARNING msg which can't be silenced using -q option.
+				out, err := ssh.Output()
+				if err != nil {
+					ginkgo.By(fmt.Sprintf("ssh attempt No.%d to node %d failed: %v", retry, worker, err.Error()))
+					time.Sleep(10 * time.Second)
+				} else {
+					buf := fmt.Sprintf("%s on Node %d", hdr, worker)
+					result[buf] = strings.Split(strings.TrimSpace(string(out)), "\n")
+					break
+				}
+			}
+			if retry >= 4 {
+				ginkgo.Fail(fmt.Sprintf("repeated ssh attempts to node %d failed", worker))
+			}
```

Comment:
We get no explanation here why it fails. Having to repeat the comparison against 4 here also isn't nice (what if we reduce it above to 3 and forget here?!).

A better retry loop is:
```
i := 0
for {
    ssh := exec.Command(sshcmd, cmd)
    out, err := ssh.Output()
    if err != nil {
        if i >= 4 {
           gingko.Fail(fmt.Sprintf("repeated ssh attempts to node %d failed again: %v", worker, err))
	}
	
         ginkgo.By(fmt.Sprintf("ssh attempt No.%d to node %d failed: %v", retry, worker, err.Error()))
         time.Sleep(10 * time.Second)
   } else {
        ...
        break
   }
   i++
}